### ISSUE TITLE
Kernel dies while converting geometry to EPSG3857, OS X 11.6

### ISSUE THREAD

**Issue by rrpelgrim** [pyproj4/pyproj#730]
I am attempting to install the latest version of pyproj in order resolve the same issue as described [in this closed issue 1113](https://github.com/geopandas/geopandas/issues/1113)

#### Installation method/steps
- installed pyproj through conda in terminal, see below.

```
(base) Richard-Pelgrims-MacBook-Pro:site-packages data_science$ conda install pyproj
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/anaconda3

  added / updated specs:
    - pyproj


The following NEW packages will be INSTALLED:

  pyproj             pkgs/main/osx-64::pyproj-1.9.6-py37h9c430a6_0
```

#### Environment Information
 - pyproj version you are attempting to install: 2.6 to resolve issues mentioned in [this issue](https://github.com/geopandas/geopandas/issues/1113)
 - PROJ version : Rel. 5.2.0, September 15th, 2018
 - Python version: 3.7.4
 - Operation System Information: Darwin-15.6.0-x86_64-i386-64bit

**Comment by snowman2** [comment]
https://pyproj4.github.io/pyproj/stable/installation.html

I would recommend creating a new conda environment as your current version of PROJ is not compatible with pyproj 2+ (see compatibility matrix in installation instructions).

```
conda create -n pyprojenv -c conda-forge pyproj
conda activate pyprojenv
```
`~/.condarc`:
```
channels:
  - conda-forge
  - defaults
channel_priority: strict
```

**Comment by snowman2** [comment]
If you have further issues, feel free to comment or reopen.

**Comment by rrpelgrim** [comment]
Thanks for the suggestion. I tried creating a new environment as per your suggestion (which I called 'py374'), but I'm still running into issues. I can't seem to be able to reopen this issue, so will post in comments. 

The problem seems to be when I then try to install geopandas, which seems to force a downgrade to an earlier version of pyproj (1.9.6). 

To be clear, I followed the following steps:
- created a new conda environment running python 3.7.4
- installed pyproj using: conda install -c conda-forge pyproj
- installed pandas using: conda install pandas
- tried to install geopandas using: conda install geopandas

`(py374) Richard-Pelgrims-MacBook-Pro:~ data_science$ conda install geopandas
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/anaconda3/envs/py374

  added / updated specs:
    - geopandas


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.1.1h             |       haf1e3a3_0         2.2 MB
    ------------------------------------------------------------
                                           Total:         2.2 MB

The following NEW packages will be INSTALLED:

  attrs              pkgs/main/noarch::attrs-20.2.0-py_0
  bzip2              pkgs/main/osx-64::bzip2-1.0.8-h1de35cc_0
  cairo              pkgs/main/osx-64::cairo-1.14.12-hc4e6be7_4
  click              pkgs/main/noarch::click-7.1.2-py_0
  click-plugins      pkgs/main/noarch::click-plugins-1.1.1-py_0
  cligj              pkgs/main/osx-64::cligj-0.7.0-py37hecd8cb5_0
  curl               pkgs/main/osx-64::curl-7.67.0-ha441bb4_0
  expat              pkgs/main/osx-64::expat-2.2.10-hb1e8313_2
  fiona              pkgs/main/osx-64::fiona-1.8.4-py37h9a122fd_0
  fontconfig         pkgs/main/osx-64::fontconfig-2.13.0-h5d5b041_1
  freexl             pkgs/main/osx-64::freexl-1.0.5-h1de35cc_0
  gdal               pkgs/main/osx-64::gdal-2.3.3-py37hbe65578_0
  geopandas          pkgs/main/noarch::geopandas-0.6.1-py_0
  geos               pkgs/main/osx-64::geos-3.7.1-h0a44026_0
  gettext            pkgs/main/osx-64::gettext-0.19.8.1-h15daf44_3
  giflib             pkgs/main/osx-64::giflib-5.1.4-h1de35cc_1
  glib               pkgs/main/osx-64::glib-2.63.1-hd977a24_0
  hdf4               pkgs/main/osx-64::hdf4-4.2.13-h39711bb_2
  hdf5               pkgs/main/osx-64::hdf5-1.10.4-hfa1e0ec_0
  icu                pkgs/main/osx-64::icu-58.2-h0a44026_3
  json-c             pkgs/main/osx-64::json-c-0.13.1-h3efe00b_0
  kealib             pkgs/main/osx-64::kealib-1.4.7-hf5ed860_6
  libboost           pkgs/main/osx-64::libboost-1.67.0-hebc422b_4
  libdap4            pkgs/main/osx-64::libdap4-3.19.1-h3d3e54a_0
  libgdal            pkgs/main/osx-64::libgdal-2.3.3-h0950a36_0
  libgfortran        pkgs/main/osx-64::libgfortran-3.0.1-h93005f0_2
  libiconv           pkgs/main/osx-64::libiconv-1.16-h1de35cc_0
  libkml             pkgs/main/osx-64::libkml-1.3.0-hbe12b63_4
  libnetcdf          pkgs/main/osx-64::libnetcdf-4.6.1-hd5207e6_2
  libpq              pkgs/main/osx-64::libpq-11.2-h051b688_0
  libspatialindex    pkgs/main/osx-64::libspatialindex-1.9.3-h0a44026_0
  libspatialite      pkgs/main/osx-64::libspatialite-4.3.0a-h644ec7d_19
  libxml2            pkgs/main/osx-64::libxml2-2.9.10-h7cdb67c_3
  munch              pkgs/main/noarch::munch-2.5.0-py_0
  openjpeg           pkgs/main/osx-64::openjpeg-2.3.0-hb95cd4c_1
  pcre               pkgs/main/osx-64::pcre-8.44-hb1e8313_0
  pixman             pkgs/main/osx-64::pixman-0.40.0-haf1e3a3_0
  poppler            pkgs/main/osx-64::poppler-0.65.0-ha097c24_1
  poppler-data       pkgs/main/osx-64::poppler-data-0.4.10-hecd8cb5_0
  proj4              pkgs/main/osx-64::proj4-5.2.0-h0a44026_1
  rtree              pkgs/main/osx-64::rtree-0.9.4-py37_1
  shapely            pkgs/main/osx-64::shapely-1.6.4-py37he8793f5_0
  xerces-c           pkgs/main/osx-64::xerces-c-3.2.3-h48eee30_0

The following packages will be REMOVED:

  proj-7.1.0-h45baca5_1

The following packages will be UPDATED:

  ca-certificates    conda-forge::ca-certificates-2020.6.2~ --> pkgs/main::ca-certificates-2020.10.14-0
  certifi            conda-forge/osx-64::certifi-2020.6.20~ --> pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3

The following packages will be SUPERSEDED by a higher-priority channel:

  krb5                  conda-forge::krb5-1.17.1-h14dd6a4_2 --> pkgs/main::krb5-1.16.4-hddcf347_0
  libcurl            conda-forge::libcurl-7.71.1-h9bf37e3_8 --> pkgs/main::libcurl-7.67.0-h051b688_0
  openssl                                       conda-forge --> pkgs/main
  pyproj             conda-forge::pyproj-2.6.1.post1-py37h~ --> pkgs/main::pyproj-1.9.6-py37h9c430a6_0
`

**Comment by rrpelgrim** [comment]
NB: I also tried creating a new environment and installing all packages specifically through the conda-forge channel. This does not resolve the issue but leads to multiple conflicts / incompatible packages. The output from Terminal is extremely long but here are the first few lines.

`(py374-forge) Richard-Pelgrims-MacBook-Pro:~ data_science$ conda install -c conda-forge geopandas
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                    |

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - pip -> python[version='>=3.9,<3.10.0a0']
  - python-dateutil -> python[version='>=3.8,<3.9.0a0']
  - wheel -> python[version='>=3.9,<3.10.0a0']

Your python: python=3.7.4

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with a past
explicit spec that is not an explicit spec in this operation (pyproj):

  - geopandas -> mapclassify -> pandas[version='>=1.0']
  - geopandas -> pandas[version='>=0.23']
  - geopandas -> pyproj[version='>=1.9,<2|>=2.2.0']

The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

(...)
Loooong list of incompatible packages


`

**Comment by snowman2** [comment]
If you are using geopandas, it has pyproj and other geospatial dependencies, so I would create a new environment just for geopandas:
```
conda create -n geo -c conda-forge geopandas
conta activate geo
(geo)
```

**Comment by rrpelgrim** [comment]
Thanks. I now managed to install geopandas with the correct PROJ and PYPROJ versions in the dedicated 'geo' environment. However, when I go back to my Jupyter Notebook and switch to the 'geo' kernel, the same problem persists: kernel dies when trying to convert geometry to EPSG3857. Also after full system restart.

output from 'conda list' in geo environment below. Pyproj is version 2.6.1.post1 / Proj is version 7.1.1

`# packages in environment at /opt/anaconda3/envs/geo:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0           py38h32f6830_1002    conda-forge
argon2-cffi               20.1.0           py38h4d0b108_2    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     20.2.0             pyh9f0ad1d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
bleach                    3.2.1              pyh9f0ad1d_0    conda-forge
boost-cpp                 1.74.0               he5d75e3_0    conda-forge
bzip2                     1.0.8                haf1e3a3_3    conda-forge
c-ares                    1.16.1               haf1e3a3_3    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
cairo                     1.16.0            h360c52f_1006    conda-forge
certifi                   2020.6.20        py38h5347e94_2    conda-forge
cffi                      1.14.3           py38h9edaa1b_1    conda-forge
cfitsio                   3.470                h4472c5d_7    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.0                      py_0    conda-forge
curl                      7.71.1               hcb81553_8    conda-forge
cycler                    0.10.0                     py_2    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
entrypoints               0.3             py38h32f6830_1002    conda-forge
expat                     2.2.9                hb1e8313_2    conda-forge
fiona                     1.8.17           py38hcfd527e_1    conda-forge
fontconfig                2.13.1            h79c0d67_1002    conda-forge
freetype                  2.10.4               h3f75d11_0    conda-forge
freexl                    1.0.5             h0b31af3_1002    conda-forge
gdal                      3.1.4            py38hdf98abe_0    conda-forge
geopandas                 0.8.1                      py_0    conda-forge
geos                      3.8.1                h4a8c4bd_0    conda-forge
geotiff                   1.6.0                hc1004f4_3    conda-forge
gettext                   0.19.8.1          haf92f58_1004    conda-forge
giflib                    5.2.1                hbcb3906_2    conda-forge
glib                      2.66.2               h4961a3e_0    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.6          nompi_h2ccf146_1110    conda-forge
icu                       67.1                 hb1e8313_0    conda-forge
importlib-metadata        2.0.0                      py_1    conda-forge
importlib_metadata        2.0.0                         1    conda-forge
ipykernel                 5.3.4            py38h1cdfbd6_1    conda-forge
ipython                   7.19.0           py38h9bb44b7_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.2           py38h32f6830_1    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
jpeg                      9d                   hbcb3906_0    conda-forge
json-c                    0.13.1            h575e443_1002    conda-forge
jsonschema                3.2.0                      py_2    conda-forge
jupyter_client            6.1.7                      py_0    conda-forge
jupyter_core              4.6.3            py38h32f6830_2    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
kealib                    1.4.13               h40102fb_1    conda-forge
kiwisolver                1.3.1            py38h9250c2b_0    conda-forge
krb5                      1.17.1               h75d18d8_3    conda-forge
lcms2                     2.11                 h174193d_0    conda-forge
libblas                   3.9.0                2_openblas    conda-forge
libcblas                  3.9.0                2_openblas    conda-forge
libcurl                   7.71.1               h9bf37e3_8    conda-forge
libcxx                    11.0.0               h439d374_0    conda-forge
libdap4                   3.20.6               h993cace_1    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.2.1             hb1e8313_1007    conda-forge
libgdal                   3.1.4                hd7bf8dc_0    conda-forge
libgfortran               5.0.0               h7cc5361_13    conda-forge
libgfortran5              9.3.0               h7cc5361_13    conda-forge
libglib                   2.66.2               h2fc9090_0    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
libkml                    1.3.0             h8ca2c65_1012    conda-forge
liblapack                 3.9.0                2_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h4844aed_106    conda-forge
libnghttp2                1.41.0               h7580e61_2    conda-forge
libopenblas               0.3.12          openmp_h54245bb_1    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libpq                     12.3                 haa216e0_2    conda-forge
libsodium                 1.0.18               haf1e3a3_1    conda-forge
libspatialindex           1.9.3                h4a8c4bd_3    conda-forge
libspatialite             5.0.0                hc85de81_0    conda-forge
libssh2                   1.9.0                h8a08a2b_5    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libwebp-base              1.1.0                hbcb3906_3    conda-forge
libxml2                   2.9.10               h2c6e4a5_2    conda-forge
llvm-openmp               11.0.0               h73239a0_1    conda-forge
lz4-c                     1.9.2                hb1e8313_3    conda-forge
markupsafe                1.1.1            py38h94c058a_2    conda-forge
matplotlib                3.3.2            py38h32f6830_1    conda-forge
matplotlib-base           3.3.2            py38had0acaf_1    conda-forge
mistune                   0.8.4           py38h4d0b108_1002    conda-forge
munch                     2.5.0                      py_0    conda-forge
nbclient                  0.5.1                      py_0    conda-forge
nbconvert                 6.0.7            py38h32f6830_2    conda-forge
nbformat                  5.0.8                      py_0    conda-forge
ncurses                   6.2                  hb1e8313_2    conda-forge
nest-asyncio              1.4.2              pyhd8ed1ab_0    conda-forge
notebook                  6.1.4            py38h32f6830_1    conda-forge
numpy                     1.19.2           py38ha98150c_1    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.3.1                h254dc36_3    conda-forge
openssl                   1.1.1h               haf1e3a3_0    conda-forge
packaging                 20.4               pyh9f0ad1d_0    conda-forge
pandas                    1.1.4            py38hcf432d8_0    conda-forge
pandoc                    2.11.0.4             h22f3db7_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.7.1              pyh9f0ad1d_0    conda-forge
pcre                      8.44                 hb1e8313_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.0.1            py38hef457fe_0    conda-forge
pip                       20.2.4                     py_0    conda-forge
pixman                    0.38.0            h01d97ff_1003    conda-forge
poppler                   0.89.0               h3232a60_1    conda-forge
poppler-data              0.4.10                        0    conda-forge
postgresql                12.3                 hb1f69c7_2    conda-forge
proj                      7.1.1                h45baca5_3    conda-forge
prometheus_client         0.8.0              pyh9f0ad1d_0    conda-forge
prompt-toolkit            3.0.8                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.7.2                      py_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.1.post1      py38h4367d1d_3    conda-forge
pyrsistent                0.17.3           py38h4d0b108_1    conda-forge
python                    3.8.6           hcfdab8c_0_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.4             pyhd8ed1ab_0    conda-forge
pyzmq                     19.0.2           py38h2c785a9_2    conda-forge
readline                  8.0                  h0678c8f_2    conda-forge
rtree                     0.9.4            py38h08f867b_1    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                49.6.0           py38h5347e94_2    conda-forge
shapely                   1.7.1            py38h8918236_1    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.33.0               h960bd1c_1    conda-forge
terminado                 0.9.1            py38h32f6830_1    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tiledb                    2.1.2                h73727fa_0    conda-forge
tk                        8.6.10               hb0a8c7a_1    conda-forge
tornado                   6.1              py38h7e3306e_0    conda-forge
traitlets                 5.0.5                      py_0    conda-forge
tzcode                    2020a                h0b31af3_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
xerces-c                  3.2.3                h0a0444a_1    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
zeromq                    4.3.3                hb1e8313_2    conda-forge
zipp                      3.4.0                      py_0    conda-forge
zlib                      1.2.11            h7795811_1010    conda-forge
zstd                      1.4.5                h289c70a_2    conda-forge
`

**Comment by snowman2** [comment]
Are you able to provide the code snipped and the dataset you are using to reproduce the issue?

Also, what is the output of `python -m pyproj -v`?

**Comment by rrpelgrim** [comment]
The code is in [this Jupyter Notebook](https://github.com/rrpelgrim/springboard_repo/blob/master/capstones/capstone-two/notebooks/01%20Data%20Collection.ipynb) under header #3 ("Importing Station Metadata") and the dataset (9.9MB) is available [here](https://drive.google.com/file/d/1njeLAUB7xpk1H_QiVgOx2fKyY2a-x-de/view?usp=sharing).

Output of python -m pyproj -v is:

(geo) Richard-Pelgrims-MacBook-Pro:~ data_science$ python -m pyproj -v
pyproj info:
    pyproj: 2.6.1.post1
      PROJ: 7.1.1
  data dir: /opt/anaconda3/envs/geo/share/proj

System:
    python: 3.8.6 | packaged by conda-forge | (default, Oct  7 2020, 18:42:56)  [Clang 10.0.1 ]
executable: /opt/anaconda3/envs/geo/bin/python
   machine: macOS-10.11.6-x86_64-i386-64bit

Python deps:
       pip: 20.2.4
setuptools: 49.6.0.post20201009
    Cython: None

**Comment by rrpelgrim** [comment]
PS I've opened [a StackOverflow question](https://stackoverflow.com/questions/64663553/kernel-dying-when-executing-to-crs-to-convert-geometry-of-geodataframe) containing minimal reproducible code.

**Comment by snowman2** [comment]
When your environment is active, what is the value for the environment variable `PROJ_NETWORK`?

**Comment by rrpelgrim** [comment]
How do I read out that value?

**Comment by rrpelgrim** [comment]
`conda env config vars list ` returns nothing in terminal

**Comment by snowman2** [comment]
`env | grep PROJ_NETWORK` should do the trick.

**Comment by rrpelgrim** [comment]
Output below. Kernel still dies with to_crs after running this. 

(geo) Richard-Pelgrims-MacBook-Pro:~ data_science$ env | grep PROJ_NETWORK
PROJ_NETWORK=ON

**Comment by snowman2** [comment]
Can you run `unset PROJ_NETWORK` in the terminal and then restart your notebook?

**Comment by rrpelgrim** [comment]
Just tried it. Kernel still dying...

**Comment by snowman2** [comment]
What happens when you run it outside of juypyter?

**Comment by rrpelgrim** [comment]
### Script ('tocrs.py') contains: 

#!/usr/bin/env python3

import pandas as pd
import geopandas as god
from shapely.geometry import Point, Polygon

df = pd.DataFrame(
    {'City': ['A', 'B', 'C', 'D', 'E'],
     'Country': ['1', '2', '3', '4', '5'],
     'Latitude': [-34.58, -15.78, -33.45, 4.60, 10.48],
     'Longitude': [-58.66, -47.91, -70.66, -74.08, -66.86]})

crs = {'init': 'epsg:4326'}

gdf = gpd.GeoDataFrame(
    df, crs=crs, geometry=gpd.points_from_xy(df.Longitude, df.Latitude))

gdf.geometry = gdf.geometry.to_crs(epsg=3857)


### Output from terminal when running script containing minimal reproducible code:

(geo) RichardgrimsMBP:desktop data_science$ python tocrs.py
Traceback (most recent call last):
  File "tocrs.py", line 4, in <module>
    import geopandas as god
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/geopandas/__init__.py", line 7, in <module>
    from geopandas.io.file import _read_file as read_file  # noqa
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/geopandas/io/file.py", line 7, in <module>
    import fiona
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/__init__.py", line 86, in <module>
    from fiona.collection import BytesCollection, Collection
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/collection.py", line 11, in <module>
    from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
ImportError: dlopen(/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/ogrext.cpython-38-darwin.so, 2): Symbol not found: _CCCryptorGCMFinalize
  Referenced from: /opt/anaconda3/envs/geo/lib/libtiledb.dylib (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libSystem.B.dylib
 in /opt/anaconda3/envs/geo/lib/libtiledb.dylib

**Comment by rrpelgrim** [comment]
### Script (tocrs.py) contains:

#!/usr/bin/env python3

import pandas as pd
import geopandas as gpd
from shapely.geometry import Point, Polygon

df = pd.DataFrame(
    {'City': ['A', 'B', 'C', 'D', 'E'],
     'Country': ['1', '2', '3', '4', '5'],
     'Latitude': [-34.58, -15.78, -33.45, 4.60, 10.48],
     'Longitude': [-58.66, -47.91, -70.66, -74.08, -66.86]})

crs = {'init': 'epsg:4326'}

gdf = gpd.GeoDataFrame(
    df, crs=crs, geometry=gpd.points_from_xy(df.Longitude, df.Latitude))

gdf.geometry = gdf.geometry.to_crs(epsg=3857)

### Output from terminal:

(geo) RichardgrimsMBP:desktop data_science$ python tocrs.py
Traceback (most recent call last):
  File "tocrs.py", line 4, in <module>
    import geopandas as gpd
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/geopandas/__init__.py", line 7, in <module>
    from geopandas.io.file import _read_file as read_file  # noqa
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/geopandas/io/file.py", line 7, in <module>
    import fiona
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/__init__.py", line 86, in <module>
    from fiona.collection import BytesCollection, Collection
  File "/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/collection.py", line 11, in <module>
    from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
ImportError: dlopen(/opt/anaconda3/envs/geo/lib/python3.8/site-packages/fiona/ogrext.cpython-38-darwin.so, 2): Symbol not found: _CCCryptorGCMFinalize
  Referenced from: /opt/anaconda3/envs/geo/lib/libtiledb.dylib (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libSystem.B.dylib
 in /opt/anaconda3/envs/geo/lib/libtiledb.dylib

**Comment by snowman2** [comment]
That is more informative. Looks like an issue with fiona.
- https://github.com/conda-forge/fiona-feedstock/
- https://github.com/toblerity/fiona

**Comment by rrpelgrim** [comment]
Alright, I will take my question there. Thanks so much for your time! 🙏

**Comment by jorisvandenbossche** [comment]
@rrpelgrim additional comment regarding the use of notebooks with kernels and conda environments (which might explain why it crashes there and not in the console): if you have jupyter notebook/lab installed in the base environment, and then use a notebook with a kernel using a different conda environment, you might need to set the datadir manually. I often do this in my notebooks locally when I use a different kernel:

```
import pyproj
pyproj.datadir.set_data_dir("/home/joris/miniconda3/envs/geo-dev/share/proj/")
```

(it might be this is only an issue when your base environment also has PROJ installed, but with an incompatible version as the one in the other env, not sure though)

**Comment by rrpelgrim** [comment]
@jorisvandenbossche thanks for the tip. When I run `pyproj.datadir.set_data_dir("/Users/data_science/.conda/envs/geo/share/proj/")` in my notebook, I get the following error `AttributeError: module 'pyproj.datadir' has no attribute 'set_data_dir'`

**Comment by rrpelgrim** [comment]
@jorisvandenbossche same error when I run it in a python terminal.

**Comment by jorisvandenbossche** [comment]
That might be because you have pyproj 1.9.6 ? (you mentioned before that this happened in some cases; the `pyproj.datadir.set_data_dir()` was added since 2.0)

**Comment by rrpelgrim** [comment]
Correct. It works when I switch to the conda environment ("geo") containing PyProj 2.6 in my Terminal.
However, it still doesn't work in my Jupyter Notebook, even though it says it's running on the Python (Geo) kernel that I created from the "geo" conda environment. Which leads me to think it's not actually running in the "geo" environment. 

Any helpful pointers here? Sorry, I'm quite new to all this, really appreciate the help!

**Comment by jorisvandenbossche** [comment]
You can check in the notebook which python is loaded. Eg `import sys; sys.prefix` or `import pyproj; pyproj.__path__`

**Comment by rrpelgrim** [comment]
Thanks. I managed to recreate the "geo" kernel and am able to execute pyproj.datadir.set_data_dir().
However, now my Notebook is reproducing the error I was getting in my terminal (in the "geo" environment containing PyProj 2.6): ImportError with Fiona. I've taken that up with the people from Fiona [here](https://github.com/conda-forge/fiona-feedstock/issues/172).

**Comment by jorisvandenbossche** [comment]
OK, at least it's good to know it's the same fiona issue. Hopefully that can be solved quickly (and it seems there is a workaround available, looking at the linked issue)

**Comment by rrpelgrim** [comment]
Problem resolved for now, I have pinned tiledb version 2.0.8 (instead of 2.1.2). Kernel no longer dying when running to_crs. Thanks so much for your time 🙏
